client/setup.go

```diff
@@ -37,10 +35,44 @@ func NewClient() (*Client, error) {
 	sc := make(chan os.Signal, 1)
 
 	c := &Client{
-		TweetBot:   client,
-		DiscordBot: dgclient,
-		ShutDown:   sc,
+		DiscordBot:     dgclient,
+		ShutDown:       sc,
+		TwitterClients: tClients,
 	}
 
 	return c, nil
 }
+
+// SetupTwitterClients will parse the local json file with twitter credentials and setup
+// clients that connect to the twitter api. jsonFileName is an optional commandline flag.
+func SetupTwitterClients(jsonFileName string) (map[string]*twitter.Client, error) {
+
+	var credentials struct {
+		ApiCredentials []struct {
+			TwitterHandle       string `json:"twitter_handle"`
+			OauthConsumerKey    string `json:"consumer_key"`
+			OauthConsumerSecret string `json:"consumer_secret"`
+			OauthAccessToken    string `json:"access_token"`
+			OauthAccessSecret   string `json:"access_secret"`
+		} `json:"twitter_clients"`
+	}
+
+	f, err := os.Open(jsonFileName)
```

Comment:
not super important, but it's a good habit to make sure you close file handles when you open them